client/src/components/ProductDetailNavBar/index.tsx

```diff
@@ -0,0 +1,90 @@
+import styled from '@emotion/styled';
+import React from 'react';
+
+export interface IProductDetailNavBar {
+  activeInfo: number;
+  setActiveInfo: (activeInfo: number) => void;
+  navMenus: string[];
+}
+
+const ProductDetailNavBar = (props: IProductDetailNavBar): JSX.Element => {
+  const { activeInfo, setActiveInfo, navMenus } = props;
+
+  return (
+    <Wrapper>
+      <Nav>
+        {navMenus.map((navMenu, i) => {
+          return (
+            <NavItem active={i === activeInfo} key={`nav-${i}`}>
+              <ItemButton
+                onClick={(e) => {
+                  setActiveInfo(i);
+                }}
+              >
```

Comment:
```suggestion
export interface IProductDetailNavBar {
  onClickNav: (activeInfo: number) => void; // setActiveInfo
  navMenus: string[];
}

const ProductDetailNavBar = (props: IProductDetailNavBar): JSX.Element => {
  const { onClickNav, navMenus } = props;
  const [activeInfo, setActiveInfo] = useState<number>(0);

  const onClick = (e: React.MouseEvent) => {
    const { index } = (e.target as HTMLElement).dataset;

    if (!index) return;
    onClickNav(+index);
    setActiveInfo(+index);
  };

  return (
    <Wrapper>
      <Nav onClick={onClick}>
        {navMenus.map((navMenu, i) => (
          <NavItem active={i === activeInfo} key={`nav-${i}`} data-index={i}>
            <ItemButton>{navMenu}</ItemButton>
          </NavItem>
        ))}
```

이런 식으로 바꿔보는 건 어떠신가요?

이렇게 바꿨을 때의 장점은,
- 이벤트 위임으로 메모리상 이벤트가 조금이나마 적어진다.
- 내부에서 `activeInfo`를 관리하기 때문에, `jest`를 통해 클릭 시 스타일이 변경되는지 확인할 수 있다.
- 외부에서 `props`로 넘겨줘야할 값이 무엇인지 명확해진다.
- 스토리북에서도 직접 눌러볼 수 있다.

정도가 될 것 같습니다!

승보님의 생각은 어떠신가요? ☺️